aiida_quantumespresso/calculations/matdyn.py

```diff
@@ -1,121 +1,72 @@
-                                  # this is dynamically added in the _prepare_for_submission
-                                  #('INPUT','flfrc',Q2rCalculation.FORCE_CONSTANTS_NAME), # input
-                                 ]
-    
-        self._internal_retrieve_list = [self._PHONON_FREQUENCIES_NAME, 
-                                        self._PHONON_DOS_NAME]
-        
-        # Default Matdyn output parser provided by AiiDA
-        self._default_parser = 'quantumespresso.matdyn'
-    
-    @classproperty
-    def _use_methods(cls):
-        """
-        Use_* methods for the matdyn class.
-        """
-        retdict = NamelistsCalculation._use_methods
-        retdict.update({
-            "kpoints": {
-               'valid_types': (KpointsData),
-               'additional_parameter': None,
-               'linkname': 'kpoints',
-               'docstring': ("Kpoints on which to calculate the phonon "
-                             "frequencies"),
-               },                        
-            })
-        return retdict
+    _PHONON_FREQUENCIES_NAME = 'phonon_frequencies.dat'
+    _PHONON_MODES_NAME = 'phonon_displacements.dat'
+    _PHONON_DOS_NAME = 'phonon_dos.dat'
 
-    def use_parent_calculation(self, calc):
-        """Set the parent calculation."""
-        if not isinstance(calc, Q2rCalculation):
-            raise ValueError('Parent calculation must be a Q2rCalculation')
+    _default_namelists = ['INPUT']
+    _blocked_keywords = [
+        ('INPUT', 'flfrq', _PHONON_FREQUENCIES_NAME),  # output frequencies
+        ('INPUT', 'flvec', _PHONON_MODES_NAME),  # output displacements
+        ('INPUT', 'fldos', _PHONON_DOS_NAME),  # output density of states
+        ('INPUT', 'q_in_cryst_coord', True),  # kpoints always in crystal coordinates
+    ]
 
-        try:
-            remote_folder = calc.get_outgoing(node_class=RemoteData, link_label_filter='remote_folder').one().node
-        except ValueError:
-            raise exceptions.UniquenessError('Parent calculation does not have a remote folder output node.')
+    _internal_retrieve_list = [
+        _PHONON_FREQUENCIES_NAME,
+        _PHONON_DOS_NAME
+    ]
+    _default_parser = 'quantumespresso.matdyn'
 
-        self.use_parent_folder(remote_folder)
-   
-    def _get_following_text(self, inputdict, settings):
-        """
-        Add the kpoints after the namelist.
-        
-        This function should consume the content of inputdict (if it requires
-        a different node) or the keys inside settings, using the 'pop' method,
-        so that inputdict and settings should remain empty at the end of 
-        _prepare_for_submission, if all flags/nodes were recognized
-        """
-        from aiida.common import InputValidationError
-        
+    @classmethod
+    def define(cls, spec):
+        super(MatdynCalculation, cls).define(spec)
+        spec.input('parent_folder', valid_type=ForceconstantsData, required=True)
```

Comment:
Is it appropriate to call it 'parent_folder' when `ForceconstantsData` is a subclass of `SinglefileData`?

Reply:
No it is not really, I agree, but changing it would be a backwards incompatible change. I would still be in favor of doing it, but that should be done in a separate PR most likely.

Reply:
Ok, but I would do it before the release of 3.0.0, since we are already making a new major release.

Reply:
I agree, let's see if we can convince Gio of this as well